fireshape/zoo/levelset_functional.py

```diff
@@ -0,0 +1,18 @@
+import firedrake as fd
+from ..objective import Objective
+
+
+__all__ = ["LevelsetFunctional"]
+
+
+class LevelsetFunctional(Objective):
+
+    def __init__(self, f, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.f = f
+
+    def value_form(self):
+        return self.f * fd.dx(domain=self.Q.mesh_m)
```

Comment:
why do we need to specify `domain=self.Q.mesh_m` on `fd.dx`?

Reply:
Because `f` might just be an expression and firedrake does not know where to integrate it if we don't tell him.